lazy_load_symbols/load_sym_lazily.py

```diff
@@ -0,0 +1,221 @@
+from __future__ import absolute_import, print_function
+
+import re
+import subprocess
+from subprocess import PIPE
+import datetime
+from collections import namedtuple
+import six
+
+import gdb
+
+PROFILING = False
+
+
+def parse_raw_symlist(syml):
+    symre = r'[0-9 ]+\s[\w]\s(.*)'
+    symc = re.compile(symre)
+    for line in syml.split('\n'):
+        m = symc.search(line)
+        if m:
+            yield m.group(1)
+
+
+def extract_symbol_list_from_so(file_name):
+    try:
+        symlist = subprocess.check_output(['nm', '-C', file_name], stderr=PIPE)
+        if not symlist:
+            pdb_name = file_name.replace('.so', '.pdb')
+            symlist = subprocess.check_output(['nm', '-C', pdb_name], stderr=PIPE)
+    except subprocess.CalledProcessError:
+        print('nm -C {} failed.'.format(file_name))
+    except OSError:
+        print('no usable "nm" found, cannot continue')
+        raise
+    return symlist
+
+
+def extract_symbols_from_raw_list(rs):
+    '''
+    Saves the entire symbol names as a single long string
+    so we can say something like:
+        if symbol in symbol_list
+    '''
+    parsed_list = []
+    for sym in rs:
+        sym_name = subprocess.check_output(['c++filt', sym])
+        assert sym_name, 'c++filt didn\'t demangle {}'.format(sym)
+        parsed_list.append(sym_name)
+    return ' '.join(parsed_list)
+
+
+SOInfo = namedtuple('SOInfo', ['fullname', 'symbols', 'start', 'end', 'loaded'])
+
+
+def load_so_list():
+    '''
+    Loads the list of shared libraries and returns a dictionary
+    '''
+    solist = gdb.execute('info sharedlibrary', to_string=True)
```

Comment:
Is your goal to be independent from UDB so this can also run in GDB?
If yes, maybe state it at the beginning of the file in the docstring.

Othwerwise, it would be better to use `debugger_utils.execute_to_string`.